packages/preview-middleware-client/src/adp/controllers/AddSubpage.controller.ts

```diff
@@ -119,47 +119,56 @@ export default class AddSubpage extends BaseDialog<AddSubpageModel> {
         const navProperty = this.model.getProperty('/selectedNavigation/key');
         const navigation = this.model.getProperty('/navigationData').find((item) => (item.navProperty = navProperty));
         const targetEntitySet = navigation?.entitySet ?? '';
-        const appType = this.model.getProperty('/appType');
-        const pageType = this.model.getProperty('/pageType');
-
-        const modifiedValue =
-            appType === 'fe-v2'
-                ? {
-                      changeType: 'appdescr_ui_generic_app_addNewObjectPage',
-                      reference: this.model.getProperty('/appReference'),
-                      parameters: {
-                          parentPage: {
-                              component: pageType,
-                              entitySet: this.model.getProperty('/currentEntitySet')
-                          },
-                          childPage: {
-                              id: `ObjectPage|${navProperty}`,
-                              definition: {
-                                  entitySet: targetEntitySet,
-                                  navigationProperty: navProperty
-                              }
-                          }
-                      }
-                  }
-                : {
-                      changeType: 'appdescr_fe_addNewPage',
-                      parameters: {
-                          sourcePage: {
-                              id: this.runtimeControl.getId(),
-                              navigationSource: targetEntitySet
-                          },
-                          targetPage: {
-                              type: 'Component',
-                              id: `${targetEntitySet}ObjectPage`,
-                              name: 'sap.fe.templates.ObjectPage',
-                              routePattern: `${targetEntitySet}({key}):?query:`,
-                              settings: {
-                                  contextPath: `/${targetEntitySet}`,
-                                  editableHeaderContent: false
-                              }
-                          }
-                      }
-                  };
+
+        let modifiedValue;
+        if (this.options.pageDescriptor.appType === 'fe-v2') {
```

Comment:
Please pass app component with the change for v2 and v2 like the example for clearing the internal cache during undo/redo, otherwise duplicate change files may be created
https://github.com/SAP/open-ux-tools/blob/5e44861590abcfd6abaf11dbbb8b56cccba1e6f3/packages/preview-middleware-client/src/adp/quick-actions/common/add-new-annotation-file.ts#L140-L143

Reply:
fixed